aes/Cargo.toml

```
@@ -31,7 +31,7 @@ hazmat = [] # Expose cryptographically hazardous APIs
 
 [lints.rust.unexpected_cfgs]
 level = "warn"
-check-cfg = ["cfg(aes_compact)", "cfg(aes_force_soft)"]
+check-cfg = ["cfg(aes_compact)", "cfg(aes_force_soft)", "cfg(avx256_disable)", "cfg(avx512_disable)"]
```

Perhaps namespace these configs under `aes_*` like the others?

```suggestion
check-cfg = ["cfg(aes_compact)", "cfg(aes_force_soft)", "cfg(aes_avx256_disable)", "cfg(aes_avx512_disable)"]
```

---

Ideally, this would be handled with ["negative" target features](https://internals.rust-lang.org/t/18169), but alas. :/

---

It also may be worth to name it just `aes_disable_vaes`. It does not look like we have different target features for VAES256 and VAES512, so it may be worth to just remove the `vaes256` backend.

---

I looked into it and it looks like some CPU families support VAES256, but not VAES512 (e.g. [Zen 3](https://en.wikipedia.org/wiki/Zen_3#Improvements)).

---

I renamed the config flags with the `aes_` prefix as suggested.